src/assets/scripts/client/trafficGenerator/SpawnPatternModel.js

```diff
@@ -0,0 +1,487 @@
+     *
+     * @for SpawnPatternModel
+     * @method getRandomAirlineForSpawn
+     * @return {string}
+     */
+    getRandomAirlineForSpawn() {
+        const index = this._findRandomIndexForList(this.airlines);
+        const airlineId = this._weightedAirlineList[index];
+
+        return airlineId;
+    }
+
+    /**
+     * Returns a random whole number between the allowed min and max delay.
+     *
+     * This is the value that will be used by the `SpawnScheduler` when
+     * when creating a new spawn interval.
+     *
+     * @for SpawnPatternModel
+     * @method getRandomDelayValue
+     * @return randomDelayValue {number}
+     */
+    getRandomDelayValue() {
+        // TODO: make this method a fascade by implementing a switch to handle #method variations with other class methods
+        let targetDelayPeriod = this._maximumDelay;
+
+        if (targetDelayPeriod < this._minimumDelay) {
+            targetDelayPeriod = this._minimumDelay;
+        }
+
+        const maxDelayPeriod = targetDelayPeriod + (targetDelayPeriod - this._minimumDelay);
+
+        return _random(this._minimumDelay, maxDelayPeriod);
+    }
+
+    /**
+     * Sets #_minimumAltitude and #_maximumAltitude from a provided altitude.
+     *
+     * Altitude may be a single number or a range, expressed as: `[min, max]`.
+     * This method handles that variation and sets the class properties with
+     * the correct values.
+     *
+     * @for SpawnPatternModel
+     * @method _setMinMaxAltitude
+     * @param altitude {array|number}
+     */
+    _setMinMaxAltitude(altitude) {
+        if (_isArray(altitude)) {
+            const [min, max] = altitude;
+
+            this._minimumAltitude = min;
+            this._maximumAltitude = max;
+
+            return;
+        }
+
+        this._minimumAltitude = altitude;
+        this._maximumAltitude = altitude;
+    }
+
+    /**
+     * Abstracted boolean logic used to determine if a category is valid.
+     *
+     * @for SpawnPatternModel
+     * @method _isValidCategory
+     * @param _isValidCategory {string}
+     * @return {boolean}
+     * @private
+     */
+    _isValidCategory(category) {
+        return category === FLIGHT_CATEGORY.DEPARTURE || category === FLIGHT_CATEGORY.ARRIVAL;
+    }
+
+    /**
+     * @for SpawnPatternModel
+     * @method _calculateMinimumDelayFromSpeed
+     * @return {number}
+     * @private
+     */
+    _calculateMinimumDelayFromSpeed() {
+        if (this.speed === 0) {
+            return 0;
+        }
+
+        return Math.floor(AIRPORT_CONSTANTS.MIN_ENTRAIL_DISTANCE_NM * (TIME.ONE_HOUR_IN_SECONDS / this.speed));
+    }
+
+    /**
+     * Calculates the upper bound of the spawn delay value.
+     *
+     * @for SpawnPatternModel
+     * @method _calculateMaximumDelayFromRate
+     * @return {number}
+     * @private
+     */
+    _calculateMaximumDelayFromRate() {
+        return Math.floor(TIME.ONE_HOUR_IN_SECONDS / this.rate);
```

Comment:
To clarify this, there should be no rounding on the *calculation of the delay time*. And I only say that because it's at the expiration of that "delay time" that the new delay is calculated. So if we are rounding them all down, then we aren't outputting the aircraft we promised we would, as shown below:

Requested rate of 41acph.
Delay between each spawn: 87.(80487)repeating seconds
Floored Delay time: 87 seconds

@ 87.80487 s/ac --> 41.00acph
@ 87.00000 s/ac --> 41.38acph

It's a miniscule level of error, but unless there's an advantage to be gained, it's worth avoiding.